plugins/registry/functions.go

```diff
@@ -100,9 +108,31 @@ func pushToRegistry(appName string, tag int, imageID string, imageRepo string) e
 		return errors.New("Unable to tag image")
 	}
 
-	// For the future, we should also add the ability to create the remote repository
-	// This is only really important for registries that do not support creation on push
-	// Examples include AWS and Quay.io
+	extraTags := getRegistryPushExtraTagsForApp(appName)
+	if extraTags != "" {
+		extraTagsArray := strings.Split(extraTags, ",")
+		imagesToRemove := []string{}
+		for _, extraTag := range extraTagsArray {
+			extraTagImage := fmt.Sprintf("%s%s:%s", registryServer, imageRepo, extraTag)
+			imagesToRemove = append(imagesToRemove, extraTagImage)
+			common.LogVerboseQuiet(fmt.Sprintf("Tagging %s as %s in registry format", imageRepo, extraTag))
+			if !dockerTag(imageID, extraTagImage) {
+				return errors.New(fmt.Sprintf("Unable to tag image as %s", extraTag))
+			}
+			func() {
+				defer func() {
+					common.LogVerboseQuiet(fmt.Sprintf("Untagging extra tag %s", extraTag))
+					if !common.RemoveImages(imagesToRemove) {
```

Comment:
`common.RemoveImages ` returns an error, and thats what you need to handle. This won't compile otherwise. What you want is:

```go
if err := common.RemoveImages(extraTagImage); err != nil {
    common.LogWarn(fmt.Sprintf("Unable to untag extra tag %s: %s", extraTag, err.Error()))
}
```

Additionally, since we're now immediately deferring the removal, you can remove any references to `imagesToRemove`.